src/oscar/templatetags/sorting_tags.py

```diff
@@ -21,45 +22,27 @@
 }
 
 
-def anchor(parser, token):
-    bits = token.split_contents()
-    if len(bits) < 2:
-        raise template.TemplateSyntaxError(
-            "anchor tag takes at least 1 argument")
-    try:
-        title = bits[2]
-    except IndexError:
-        title = bits[1].capitalize()
-    return SortAnchorNode(bits[1].strip(), title.strip())
-
-
-class SortAnchorNode(template.Node):
-    def __init__(self, field, title):
-        self.field = template.Variable(field)
-        self.title = template.Variable(title)
-
-    def render(self, context):
-        field = self.field.resolve(context)
-        title = self.title.resolve(context)
-
-        request = context['request']
-        get_vars = request.GET.copy()
-        sort_field = get_vars.pop('sort', [None])[0]
-
-        icon = ''
-        if sort_field == field:
-            # We are already sorting on this field, so we set the inverse
-            # direction within the GET params that get used within the href.
-            direction = get_vars.pop('dir', [''])[0]
-            get_vars['dir'] = sort_directions[direction]['inverse']
-            icon = sort_directions[direction]['icon']
-
-        href = '%s?sort=%s' % (request.path, field)
-        if len(get_vars) > 0:
-            href += "&%s" % get_vars.urlencode()
-        if icon:
-            title = "%s %s" % (title, icon)
-        return '<a href="%s">%s</a>' % (href, title)
-
-
-anchor = register.tag(anchor)
+@register.simple_tag(takes_context=True)
+def anchor(context, field, title=None):
+    field = field.strip()
+    if title is None:
+        title = field.capitalize()
+    title = title.strip()
+    request = context['request']
+    get_vars = request.GET.copy()
+    sort_field = get_vars.pop('sort', [None])[0]
+
+    icon = ''
+    if sort_field == field:
+        # We are already sorting on this field, so we set the inverse
+        # direction within the GET params that get used within the href.
+        direction = get_vars.pop('dir', [''])[0]
+        get_vars['dir'] = sort_directions[direction]['inverse']
+        icon = sort_directions[direction]['icon']
+
+    href = '%s?sort=%s' % (request.path, field)
+    if len(get_vars) > 0:
+        href += "&%s" % get_vars.urlencode()
+    if icon:
+        title = "%s %s" % (title, icon)
+    return mark_safe('<a href="%s">%s</a>' % (href, title))
```

Comment:
It's hard to imagine, but could we somehow accidentally end up with an unsafe `title`?

Reply:
Seems unlikely, but I've added escaping for it (and included a check in the tests).